geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearDUnitTest.java

```diff
@@ -330,6 +337,101 @@ public void normalClearFromAccessorWithoutWriterButWithWriterOnDataStore() {
         .isEqualTo(0);
   }
 
+  @Test
+  public void normalClearFromDataStoreUpdatesStats() {
+    configureServers(false, true);
+    client1.invoke(this::initClientCache);
+    client2.invoke(this::initClientCache);
+
+    // Verify no clears have been recorded in stats
+    dataStore1.invoke(() -> {
+      PartitionedRegion region = (PartitionedRegion) getRegion(false);
+
+      for (BucketRegion bucket : region.getDataStore().getAllLocalBucketRegions()) {
+        long clearCount = bucket.getCachePerfStats().getRegionClearCount();
+        assertThat(clearCount).isEqualTo(0);
+      }
+    });
+
+    accessor.invoke(() -> feed(false));
+    verifyServerRegionSize(NUM_ENTRIES);
+    dataStore1.invoke(() -> getRegion(false).clear());
+    verifyServerRegionSize(0);
+
+    // Verify the stats were properly updated for the bucket regions
+    dataStore1.invoke(() -> {
+      PartitionedRegion region = (PartitionedRegion) getRegion(false);
+      long clearCount = 0L;
```

Comment:
Rather than just comparing the clear counts of the buckets between eachother, it might be better to compare all of them to the expected clear count. The current check just asserts that the clear counts for all the buckets are equal, but not that that value is necessarily correct.

Reply:
The trick here is that we don't know exactly how many buckets will be in each datastore for sure. In the current iteration of the test, 2 members will have 7 and one member will have 6 buckets (although I'd imagine you could end up with slightly different balance in specific scenarios). Originally I changed the test to have a number of total buckets across all members that was evenly divisible by the number of datastores - but this felt like a hardcoded trap for anyone trying to modify the test in the future. In the end I decided that it was good enough to determine that the count was incrementing to a consistent non-zero value even if I couldn't make sure that value matched up to a hardcoded expectation - but if you feel this is insufficient I can take another shot at this check.

Reply:
If it's safe to assume that buckets aren't moving around during the clear (which in this test seems to be the case) then it should be possible to get the number of buckets on each member with `region.getDataStore().getAllLocalBucketRegions()` and use that for verification.

Reply:
Done.